### PR TITLE
Header and footer variants behind a switch

### PR DESCRIPTION
**Pull request by zeftilldeath**
## What does this change?
To coincide with a certain campaign and to further unite print and digital the nav and footer have been redesigned.

New version of the header and footer behind a switch.
Predominantly layout and design changes to the header and footer, and a visual change to the side margins. 

Featuring:
- Some colour changes 😉 
- More prominent supporter messaging in the header and footer
- Increased prominence of supporter messaging in expanded mobile menu
- Slim nav at mobile breakpoints now contains supporter messaging (See the theme here?)
- Increased visual differentiation of Search and Account areas, including iconography
- Account more consistent across breakpoints
- Slim nav/roundel no longer overlaps page content
- Footer content has been re-ordered across editions for increased clarity
- Secondary menus carry pillar colour in footer and header
- Dated search form overrides have been updated 
- Discount codes links added to header and footer in UK edition

## Screenshots
Nope

### Does this affect other platforms?
AMP to be looked at separately

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?
Paid content is catered for in the changes.

There are some more visual changes I wish to make but I will do these in subsequent PRs.

### CODE REVIEW

**Comment by GHaberis** [comment]
@NataliaLKB @SiAdcock details of the switch can be seen in this PR that was merged into this branch: https://github.com/guardian/frontend/pull/20613

File: common/app/conf/switches/FeatureSwitches.scala [hunk 1]
```diff
@@ -516,4 +516,13 @@ trait FeatureSwitches {
     exposeClientSide = true
   )
 
+  val NewNavEnabled = Switch(
+    SwitchGroup.Feature,
+    "new-nav-enabled",
+    "If this is switched on then the new navigation will be enabled",
+    owners = Seq(Owner.withGithub("GHaberis")),
+    safeState = Off,
+    sellByDate = never,
```
Comment (SiAdcock) [inline common/app/conf/switches/FeatureSwitches.scala]: Do we want to stick a date on this to ~annoy the hell out of us~ remind us to tidy up this code?

File: common/app/views/fragments/header.scala.html [hunk 2]
```diff
@@ -36,7 +36,98 @@
...
+                    }
+                </div>
+
+                <div class="new-header__top-bar hide-until-mobile">
+                    @defining(Edition(request).id.toLowerCase()) { editionId =>
+                        @if(!page.metadata.hasSlimHeader) {
+                            @if(editionId != "au" || editionId == "uk") {
+                                <div class="top-bar__commercial-items">
+                                    <span class="top-bar__item__seperator hide-until-desktop"></span>
+                                    @if(editionId != "au") {
```
Comment (SiAdcock) [inline common/app/views/fragments/header.scala.html]: This will always be true I think

**Comment by NataliaLKB** [comment]
Could we rebase these commits? 52 is quite a lot ...

**Comment by zeftilldeath** [comment]
I've rebased logical duplicates, but don't know how much more I should combine now? @NataliaLKB

**Comment by nicl** [comment]
@zeftilldeath ideally we'd have a commit for the header and another for the footer, each with a concise heading and then detailed (multi-lined) description outlining the what and why of the change.

The github description can then simply mirror these.

This is my ideal anyway! :)

File: common/app/conf/switches/FeatureSwitches.scala (hunk 1 above)
Comment (NataliaLKB) [reply common/app/conf/switches/FeatureSwitches.scala]: Yes definitely! I would probably have it expire a week from Monday?

File: common/app/views/fragments/page/head/stylesheets/styles.scala.html [hunk 3]
```diff
@@ -52,6 +54,7 @@
 <!--[if (gt IE 9)|(IEMobile)]><!-->
 @styles.criticalCss
 @styles.linkCss
+@styles.footerCss
```
Comment (SiAdcock) [inline common/app/views/fragments/page/head/stylesheets/styles.scala.html]: Should these styles go outside the IE conditional statements? It looks like we'll need them on every browser?

File: static/src/stylesheets/layout/new-nav/_menu-item.scss [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+        text-decoration: underline;
+    }
+
+    > * {
+        pointer-events: none;
+    }
+
+    .menu-group--primary > *:not(:last-child) > &,
+    &[data-link-name='nav2 : The Guardian app'],
+    &[data-link-name='nav2 : facebook'],
```
Comment (SiAdcock) [inline static/src/stylesheets/layout/new-nav/_menu-item.scss]: These selectors look fragile to me. Is there no class we could hook into?

**Comment by SiAdcock** [comment]
Can we at least keep the [`Adcock lint`](https://github.com/guardian/frontend/pull/20646/commits/2eff8043583ad286f0e1b0961207aa0b94487454) commit? 😛

**Comment by zeftilldeath** [comment]
> @zeftilldeath ideally we'd have a commit for the header and another for the footer, each with a concise heading and then detailed (multi-lined) description outlining the what and why of the change.
> 
> The github description can then simply mirror these.
> 
> This is my ideal anyway! :)

I'm aware people are in the process of reviewing this. Should I rebase this PR into a footer/ header commit now? 

I thought I had provided enough detail on what's changed, although I never know how much detail to go into - especially with design changes. With regards to the why, again not sure how much goes up on here.

File: common/app/views/fragments/page/head/stylesheets/styles.scala.html (hunk 3 above)
Comment (GHaberis) [reply common/app/views/fragments/page/head/stylesheets/styles.scala.html]: This final block of 3 `@styles` links targets all browsers, the 2 blocks above are targeting IE.

**Review by NataliaLKB — APPROVED** [review]
There is a lot of code here, so I may have missed things - but I think it looks good! Thanks for all the work @zeftilldeath @GHaberis . The new design looks brilliant.

File: static/src/stylesheets/layout/new-nav/_search-dropdown.scss [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+.gstl_50, .gssb_c {
+    z-index: $zindex-main-menu + 1 !important;
+}
+
+.gsc-completion-container,
+.gsc-input {
+    font-family: $f-sans-serif-text;
+    padding: 0 !important;
```
[the remark this answers is not in the document]
Comment (zeftilldeath) [reply static/src/stylesheets/layout/new-nav/_search-dropdown.scss]: 100% agree.

**Pull request merged**